contracts/TxRelay.sol

```diff
@@ -0,0 +1,107 @@
+pragma solidity 0.4.15;
+
+
+// This contract is meant as a "singleton" forwarding contract.
+// Eventually, it will be able to forward any transaction to
+// Any contract that is built to accept it.
+contract TxRelay {
+
+    // Note: This is a local nonce.
+    // Different from the nonce defined w/in protocol.
+    mapping(address => uint) nonce;
+
+    // This mapping specifies a whitelist of allowed senders for transactions.
+    // There can be one whitelist per ethereum account, which is the owner of that
+    // whitelist. Users can specify which whitelist they want to use when signing
+    // a transaction. They can use their own whitelist, a whitelist belonging
+    // to another account, or skip using a whitelist by specifying the zero address.
+    mapping(address => mapping(address => bool)) public whitelist;
+
+    /*
+     * @dev Relays meta transactions
+     * @param sigV, sigR, sigS ECDSA signature on some data to be forwarded
+     * @param destination Location the meta-tx should be forwarded to
+     * @param data The bytes necessary to call the function in the destination contract.
+     * Note: The first encoded argument in data must be address of the signer. This means
+     * that all functions called from this relay must take an address as the first parameter.
+     */
+    function relayMetaTx(
```

Comment:
nonce needs to be provided as an argument, and checked to match the current nonce before it is incremented on L47